kernel/src/process/program_loader/elf/elf_file.rs

```diff
@@ -201,14 +206,14 @@ fn check_elf_header(elf_header: &ElfHeader) -> Result<()> {
     // if elf_header.pt1.os_abi() != header::OsAbi::SystemV {
     //     return Error::new(Errno::ENOEXEC);
     // }
-    // x86_64 architecture
-    debug_assert_eq!(elf_header.pt2.machine.as_machine(), header::Machine::X86_64);
-    if elf_header.pt2.machine.as_machine() != header::Machine::X86_64 {
-        return_errno_with_message!(Errno::ENOEXEC, "Not x86_64 executable");
+    if elf_header.pt2.machine.as_machine() != EXPECTED_ELF_MACHINE {
+        return_errno_with_message!(
+            Errno::ENOEXEC,
+            "Executable could not be run on this architecture"
```

Comment:
-> Executable file/program

Reply:
"Executable" can also be noun? I'm not sure if it's necessary to add "file"/"program".